src/consensus/encode.rs

```diff
@@ -606,7 +605,7 @@ impl_vec!(Vec<u8>);
 impl_vec!((u32, Address));
 impl_vec!(u64);
 
-impl Encodable for Vec<u8> {
```

Comment:
I'm pretty sure @TheBlueMatt deliberately got rid of this because the coercion of arrays to `[u8]` can happen in surprising circumstances.

Reply:
Yeah since we're encoding the length prefix here, I'd leave it for explicit `Vec<u8>` as well. We haven't had any problems with that. It avoids accidentally encoding arrays with a prefix, like Andrew mentions.

Reply:
Makes sense, haven't thought about it.